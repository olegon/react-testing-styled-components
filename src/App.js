import React from 'react';
import 'styled-components';
import logo from './logo.svg';
import styled, { keyframes } from 'styled-components';


// CRITICAL: Always use Styled Components outside the render method
const StyledApp = styled.div`
  text-align: center;
`

const StyledHeader = styled.div`
  background-color: ${props => props.timeOfTheDay === 'day' ? '#ac9a61' : '#282c34' };
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`

const StyledLink = styled.a`
  color: #61dafb;

  &:hover {
    color: #862b56;
  }
`

const rotateKeyFrames = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const StyledLogo = styled.img`
  height: 40vmin;
  pointer-events: none;

  // 1200px for testing...
  @media (min-width: 1200px) {
    animation: ${rotateKeyFrames} infinite 20s linear;
  }
`

function App() {
  return (
    <StyledApp>
      <StyledHeader timeOfTheDay="night" /* Testing styling props based */>
        <StyledLogo src={logo} alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <StyledLink

          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </StyledLink>
      </StyledHeader>
    </StyledApp>
  );
}

export default App;
