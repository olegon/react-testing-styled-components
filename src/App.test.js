import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components'
import App from './App';

describe('Testing the App component', () => {
  it('Testing Jest Snapshots with Styled Components', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
})