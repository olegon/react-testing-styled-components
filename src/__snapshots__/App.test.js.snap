// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App component Testing Jest Snapshots with Styled Components 1`] = `
.c0 {
  text-align: center;
}

.c1 {
  background-color: #282c34;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.c3 {
  color: #61dafb;
}

.c3:hover {
  color: #862b56;
}

.c2 {
  height: 40vmin;
  pointer-events: none;
}

@media (min-width:1200px) {
  .c2 {
    -webkit-animation: eykXYV infinite 20s linear;
    animation: eykXYV infinite 20s linear;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="logo"
        class="c2"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="c3"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </div>
  </div>
</div>
`;
